ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandStoreTest.java

```diff
@@ -100,212 +95,148 @@ public void shouldHaveAllCreateCommandsInOrder() {
   }
 
   @Test
   public void shouldCleanupCommandStatusOnProduceError() {
-    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
-
     // Given:
-    expect(commandProducer.send(anyObject())).andThrow(new RuntimeException("oops")).times(1);
-    expect(commandProducer.send(anyObject())).andReturn(future).times(1);
-    replay(commandProducer);
+    when(commandIdAssigner.getCommandId(any(Statement.class))).thenReturn(commandId1).thenReturn(commandId1);
+    doThrow(new RuntimeException("oops!")).doNothing().when(commandTopic).send(any(), any());
     try {
-      commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
-      fail("enqueueCommand should have raised an exception");
-    } catch (final KsqlException e) {
+      commandStore.enqueueCommand("foo", statement, ksqlConfig, Collections.emptyMap());
+      fail();
+    } catch (RuntimeException r) {
+      // Do nothing.
     }
 
     // When:
-    commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+    commandStore.enqueueCommand("foo", statement, ksqlConfig, Collections.emptyMap());
 
     // Then:
-    // main condition being verified is that the above call doesn't throw
-    verify(commandProducer);
+    verify(commandTopic, times(2)).send(any(CommandId.class), any(Command.class));
+
   }
 
   @Test
-  public void shouldEnqueueNewAfterHandlingExistingCommand() throws InterruptedException, ExecutionException {
-    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
-
+  public void shouldEnqueueNewAfterHandlingExistingCommand() {
     // Given:
-    setupConsumerToReturnCommand(commandId, command);
-    expect(commandProducer.send(anyObject(ProducerRecord.class))).andAnswer(
-        () -> {
-          commandStore.getNewCommands();
-          return future;
-        }
-    ).times(1);
-    expect(commandProducer.send(anyObject(ProducerRecord.class))).andReturn(future);
-    future.get();
-    expectLastCall().andStubReturn(null);
-    replay(future, commandProducer);
-    commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+    when(commandIdAssigner.getCommandId(any(Statement.class))).thenReturn(commandId1).thenReturn(commandId1);
 
     // When:
-    commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+    commandStore.enqueueCommand("foo", statement, ksqlConfig, Collections.emptyMap());
+
 
     // Then:
-    verify(future, commandProducer, commandConsumer);
+    verify(commandTopic).send(same(commandId1), any(Command.class));
+
   }
 
   @Test
-  public void shouldRegisterBeforeDistributeAndReturnStatusOnGetNewCommands()
-      throws ExecutionException, InterruptedException {
-    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
-
+  public void shouldRegisterBeforeDistributeAndReturnStatusOnGetNewCommands() {
```

Comment:
Refactored it without needing `getCommandStatus`. I had to make some changes to this tests since in this PR we are switching to use `CommandTopic` and producer and consumer objects are moved into CommandTopic class. So I updated some of the tests and moved some to the `CommandTopicTest`.